urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/ContentRange.java

```
@@ -0,0 +1,261 @@
+	
+	if( pair.getValue() == null || pair.getValue().length() == 0 )
+	    throw new MalformedRequestException( "'Content-Range' value misses byte-content-range-spec: " + headerValue + "." );
+
+	String bytesUnit = pair.getKey();
+	
+
+	// Split value at '/' into 'byte-range-resp-spec' and 'instance-length'
+	KeyValuePair<String,String> valuePair = KeyValuePair.splitLine( pair.getValue(), 
+									"/",    // separator
+									true    // trim
+									);
+	
+	if( valuePair.getKey() == null || valuePair.getKey().length() == 0 )
+	    throw new MalformedRequestException( "'Content-Range' value misses 'byte-range-resp-spec': " + headerValue + "." );
+	if( valuePair.getValue() == null || valuePair.getValue().length() == 0 )
+	    throw new MalformedRequestException( "'Content-Range' value misses 'instance-length': " + headerValue + "." );
+
+	
+	// Split 'byte-range-resp-spec' at '-' into 'first-byte-pos' and 'last-byte-pos'
+	KeyValuePair<String,String> rangePair = KeyValuePair.splitLine( valuePair.getKey(), 
+									"-",    // separator
+									true    // trim
+									);
+
+	if( rangePair.getKey() == null || rangePair.getKey().length() == 0 )
+	    throw new MalformedRequestException( "'Content-Range' value misses 'first-byte-position': " + headerValue + "." );
+	if( rangePair.getValue() == null || rangePair.getValue().length() == 0 )
+	    throw new MalformedRequestException( "'Content-Range' value misses 'last-byte-position': " + headerValue + "." );
+
+	
+	try {
+
+	    long firstBytePos = Long.parseLong( rangePair.getKey() );
+	    long lastBytePos  = Long.parseLong( rangePair.getValue() );
+
+	    long instanceLen = -1;
+	    if( !valuePair.getValue().equals("*") )
+		instanceLen = Long.parseLong( valuePair.getValue() );
+
+
+	    return new ContentRange( bytesUnit,
+				     firstBytePos,
+				     lastBytePos,
+				     instanceLen );
+
+	} catch( NumberFormatException e ) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 230.